src/services/company.js

```diff
@@ -133,6 +139,40 @@ class CompanyService {
             throw err;
         }
     }
+
+    disable(companyId, adminReason) {
+        return Company.findByIdAndUpdate(
+            companyId,
+            {
+                isDisabled: true,
+                adminReason,
+            },
+            { new: true },
+            (err) => {
+                if (err) {
+                    console.error(err);
+                    throw err;
+                }
+            });
+    }
```

Comment:
You can reuse the changeAttributes. Also, take advantage of the async nature of functions.

This change applies to the enable function as well
```suggestion
    disable(companyId, adminReason) {
    try{
        return this.changeAttributes(
            companyId,
            {
                isDisabled: true,
                adminReason,
            }
            );
            } catch(err) {
                    console.error(err);
                    throw err;
                }
    }
```